File: backend/src/main/java/com/awais/hr/module/benefits/service/BenefitsService.java
package com.awais.hr.module.benefits.service;

import java.util.List;
import java.util.Map;

public interface BenefitsService {
    List<Map<String, Object>> getPlans();
    void addPlan(Map<String, Object> body);
    List<Map<String, Object>> getMyEnrollments(String email);
    List<Map<String, Object>> getAllEnrollments();
    void enroll(String email, String planId);
    void unenroll(String email, String planId);
}

File: frontend/src/services/benefitsService.js
import apiClient from './api';

export const getPlans = () => apiClient.get('/benefits/plans');
export const addPlan = (dto) => apiClient.post('/benefits/plans', dto);
export const getMyEnrollments = () => apiClient.get('/benefits/my-enrollments');
export const getAllEnrollments = () => apiClient.get('/benefits/enrollments');
export const enroll = (planId) => apiClient.post(`/benefits/enroll/${planId}`, {});
export const unenroll = (planId) => apiClient.post(`/benefits/unenroll/${planId}`, {});
